[curiousfun88] iP

curiousfun88 opened the pull request:
# Blob!!

> “I am opposite of Artificial Intelligence. I am Genuine Stupidity.” – Unknown [(source)](https://example.com)

Blob helps you do the things that you are too lazy to do yourself. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://github.com/curiousfun88/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉  

And it is *FREE!*  

## Features:

- [x] Managing tasks  
- [ ] Managing deadlines (coming soon)  
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

curiousfun88 committed:
Level-0: Rename, Greet, Exit

curiousfun88 committed:
Level-1: Echo

curiousfun88 committed:
Level-2: Add, List

curiousfun88 committed:
Level-3: Mark as Done

curiousfun88 committed:
Level-4: ToDos, Events, Deadlines

curiousfun88 committed:
A-TextUiTesting: Automated Text UI Testing

curiousfun88 committed:
Level-5: Handle Errors

curiousfun88 committed:
Level-6: Delete

curiousfun88 committed:
Add javadoc comments

curiousfun88 committed:
Move all task operations into a separate TaskList class

curiousfun88 committed:
Add Level-7: Save edits

curiousfun88 committed:
Merge branch 'branch-Level-7'

curiousfun88 committed:
Add Level-8: Dates and Times

curiousfun88 committed:
Merge branch 'branch-Level-8'

curiousfun88 committed:
Add A-MoreOOP

curiousfun88 committed:
Merge branch 'branch-moreOOP'

curiousfun88 committed:
Add A-Packages: Organize into Packages

curiousfun88 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

curiousfun88 committed:
Add A-JUnit edits

curiousfun88 committed:
Merge branch 'branch-JUnit'

curiousfun88 committed:
Add A-Jar: Create a JAR File

curiousfun88 committed:
Add A-JavaDoc comments

curiousfun88 committed:
Add A-CodingStandard edits

curiousfun88 committed:
Add A-Javadoc comments edits

curiousfun88 committed:
Add Level-9: Find edits

curiousfun88 committed:
Merge branch 'branch-A-JavaDoc'

curiousfun88 committed:
Merge branch 'branch-A-CodingStandard'

curiousfun88 committed:
Merge branch 'branch-Level-9'

crmlatte commented on src/main/java/blob/Blob.java:
Nice! I think the java docs are good and coding standard is good

crmlatte commented on src/main/java/blob/TaskList.java:
The method and variable naming looks gd!

crmlatte commented on src/main/java/blob/Parser.java:
Maybe can use switch statement here but is easy to read too!

crmlatte commented on src/test/java/blob/StorageTest.java:
Method names are clear!

crmlatte reviewed (commented):
Overall your code looks easy to read and understand. gd job

curiousfun88 committed:
Add A-CheckStyle edits

curiousfun88 committed:
Add Level-10: GUI edits

curiousfun88 committed:
Add Level-10: GUI edits

curiousfun88 committed:
Merge branch 'branch-A-CheckStyle'

curiousfun88 committed:
Merge branch 'branch-Level-10'

curiousfun88 committed:
Add CheckStyle corrections

curiousfun88 committed:
Blob, TaskList: Add assertions for input validation and task addition

Previously, the getResponse method did not verify the validity of user input and processed command results, and the add method lacked a check to confirm that tasks were added correctly.

Adding assertions improves reliability by verifying:
* User input is not null before processing.
* The processed command result is not null.
* The task list size increases by one after adding a task.

Let's include these assertions to catch potential issues early and ensure data integrity.

curiousfun88 committed:
Add Javadoc comments for Main and utility methods
*  Added Javadoc comments to the getUserDialog and getBlobDialog methods in the DialogBox class to provide a clearer explanation of their functionality.
*  Documented the start method in the Main class to explain its role in initializing and displaying the main GUI window.
These additions improve code readability and maintainability, providing developers with a clear understanding of the methods' purposes and usage.

curiousfun88 committed:
Add A-BCD extension edits for C-DetectDuplicates

- Implemented the A-BCD extension in the respective isDuplicate(TaskType) method to detect duplicates for each task type.
- Adjusted logic to improve detection of duplicate tasks.
- Refined the process of checking task uniqueness in the task list.
- Updated method comments and documentation for clarity and maintainability.

This change improves the accuracy of detecting duplicate tasks within the task list, making the system more reliable and user-friendly.

curiousfun88 committed:
Merge pull request #1 from curiousfun88/branch-A-Assertions

Blob, TaskList: Add assertions for input validation and task addition

curiousfun88 committed:
Merge pull request #2 from curiousfun88/branch-A-CodeQuality

Add Javadoc comments for Main and utility methods

curiousfun88 committed:
Merge branch 'master' into branch-BCD-Extension

curiousfun88 committed:
Merge pull request #3 from curiousfun88/branch-BCD-Extension

Add A-BCD extension edits for C-DetectDuplicates

curiousfun88 committed:
Add A-BetterGUI and A-Personality edits

curiousfun88 committed:
Add A-UserGuide edits

curiousfun88 committed:
Add checkstyle edits

curiousfun88 committed:
Add data file edits

curiousfun88 committed:
Add Code Quality Edits

curiousfun88 committed:
Add bug correction edits

curiousfun88 committed:
Add Code Quality Edits

curiousfun88 committed:
Update README :)
